### PR TITLE
AI doc update - Select & Date picker

### LINKED ISSUE #4011: AI documentation: Add AI docs for Date Picker
Update the checkbox component page using the [AI component guidelines template](https://github.com/carbon-design-system/carbon-website/issues/4007).

## Usage tab
[See example in Figma](https://www.figma.com/file/OJwBWBlctEgm9zYUCYbhu3/Carbon-for-AI-%E2%80%93-website-integration?type=design&node-id=80%3A7220&mode=design&t=XPoeQ0K8Ea0VbhbW-1)

- [x] Add inline notification to the top of the page under intro paragraph. The link should go to the AI component’s storybook example.
- [x] Add “AI presence” to the anchor links.
- [x] Add an H2 section for “AI presence” between “Modifiers” (or “Behaviors”) and “Related” content sections.
- [x] Copy the content from the template example and edit the template content to be specific to each component. Only the input components will get the extra revert h3 sections.
- [x] Create images for the components similar to the template examples (add images to the [Usage and Style images](https://www.figma.com/file/9J3QEJ717fLUa7tzlWVMMu/Usage-and-Style-images?type=design&t=VpbAnlbxwfthmkMr-6) file)

_If links to the general AI guidelines sections aren’t ready yet then don’t include links or hyperlinks yet. We will add these in later when ready._

## Style
[See example in Figma](https://www.figma.com/file/OJwBWBlctEgm9zYUCYbhu3/Carbon-for-AI-%E2%80%93-website-integration?type=design&node-id=80%3A7192&mode=design&t=XPoeQ0K8Ea0VbhbW-1)

- [x] Add “AI presence” to the anchor links.
- [x] Add an H2 section for “AI presence” at the end of the page
- [x] Copy the content from the template example and edit the template content where needed to be specific to each component.
- [x] Create an image for the component similar to the template examples. (add images to the [Usage and Style images](https://www.figma.com/file/9J3QEJ717fLUa7tzlWVMMu/Usage-and-Style-images?type=design&t=VpbAnlbxwfthmkMr-6) file)

_If links to the general AI guidelines sections aren’t ready yet then don’t include links or hyperlinks yet. We will add these in later when ready._

## Accessibility 
No changes needed at the moment.

---

# Deliverables
- [x] Pull request opened in the website repo with content updates

### PR DESCRIPTION
**Pull request by Kritvi-bhatia17**
Closes #4011 & #4038 

#### Changelog (both components)

_Added in the Usage tab_
- Inline notification
- AI Presence Section
- AI Revert Section

_Added in the Style tab_
- AI Styling table
- AI Default and Fluid states for respective components

---

#### Changelog (only Select)

_Added in the Style tab_
- Page description + Anchor links
- Feedback section

### CODE REVIEW

**Review by tay1orjones — APPROVED** [review]
When flipping between the images, the scale and positioning shift slightly. Not sure if that's worth fixing though. Otherwise looks great to me!

**Comment by Kritvi-bhatia17** [comment]
> Visit Preview

Thanks, @tay1orjones! So, if I understand correctly, the shift you're referring to is due to the default and fluid components in the images, or something else? 🤔

**Review by aagonzales — APPROVED** [review]
No notes. Ship it!

**Comment by aagonzales** [comment]
Oh seeing, @tay1orjones comment now. We do try to align some parts of the default/fluid images so they don't bounce so much. Sometimes its aligning the bases line or sometime its the text or icons in the field. Look at the "keyboard" image under "interactions" on the select page. 

Its not a deal break but is a nice visual polish moment to look out for.

**Comment by Kritvi-bhatia17** [comment]
Hi @tay1orjones, @aagonzales, I tried fixing the positioning shift, please have a look!

File: src/pages/components/date-picker/style.mdx [hunk 1]
```diff
@@ -468,6 +469,49 @@ group. Time picker's total width will vary based on the grid and layout
 
 <Caption>Sizes for default style time picker input | px / rem</Caption>
 
+## AI presence
+
+The following are the unique styles applied to the components when the AI slug
+is present. Unless specified, all other tokens in the components remain the same
+as the default variants.
```
Comment (aagonzales) [inline src/pages/components/date-picker/style.mdx:476]: Let's change "default" here as well. 

```suggestion
as the non-AI variants.
```

File: src/pages/components/select/style.mdx [hunk 2]
```diff
@@ -216,3 +234,52 @@ and any proper nouns capitalized. Select text should be three words or less.
 </div>
 
 <Caption>Size for fluid select | px / rem</Caption>
+
+## AI presence
+
+The following are the unique styles applied to the components when the AI slug
+is present. Unless specified, all other tokens in the components remain the same
+as the default variants.
```
Comment (aagonzales) [inline src/pages/components/select/style.mdx:242]:
```suggestion
as the non-AI variants.
```

**Review by aagonzales — COMMENTED** [review]
Looks good, but I just spotted one more place on the style tab where we should swap out "default" for "non-AI"

**Review by tay1orjones — APPROVED** [review]

File: src/pages/components/date-picker/usage.mdx [hunk 3]
```diff
@@ -565,6 +575,67 @@ a meeting time.
...
+</Tabs>
+
+</Column>
+</Row>
+
+### Revert to AI
+
+A date picker can toggle between the AI variant and the non-AI variant depending
+on the user’s interaction. If the user manually overrides the AI-suggested
+content then the input will change from the AI variant to the default variant.
```
Comment (alina-jacob) [inline src/pages/components/date-picker/usage.mdx:613]: Hi @Kritvi-bhatia17 just noticed this default to non-AI variant change in the usage tab for Date picker and Select component, rest is all good! 🚀 

_"content then the input will change from the AI variant to the **non-AI** variant."_

File: src/pages/components/select/usage.mdx [hunk 4]
```diff
@@ -415,6 +426,67 @@ red border, an error icon indicator, and an error message.
...
+</Tabs>
+
+</Column>
+</Row>
+
+### Revert to AI
+
+A select can toggle between the AI variant and the non-AI variant depending on
+the user’s interaction. If the user manually overrides the AI-suggested content
+then the input will change from the AI variant to the default variant. Once
```
Comment (alina-jacob) [inline src/pages/components/select/usage.mdx:464]: _"content then the input will change from the AI variant to the **non-AI** variant."_

**Comment by Kritvi-bhatia17** [comment]
Thanks @alina-jacob! I also just noticed it while updating the other component will fix this in the upcoming PR!

File: src/pages/components/select/images/select-usage-AI-presence-default.png
Comment (alina-jacob) [inline src/pages/components/select/images/select-usage-AI-presence-default.png:1]: hi @Kritvi-bhatia17, just one more observation, the input field in this image should show some 'AI selected option'
And the menu item will subsequently reflect that (for both default and fluid states)

Here's an example from the dropdown component: 
image

Subsequently, while showing the revert image, just change the option item.

File: src/pages/components/select/images/select-usage-revert-to-AI-fluid.png
Comment (alina-jacob) [inline src/pages/components/select/images/select-usage-revert-to-AI-fluid.png:1]: Example for revert state (where user has changed the AI selected option) 

image

**Review by alina-jacob — COMMENTED** [review]
Select component: Input field suggestions

File: src/pages/components/select/images/select-usage-revert-to-AI-fluid.png
Comment (Kritvi-bhatia17) [reply src/pages/components/select/images/select-usage-revert-to-AI-fluid.png:1]: Thanks @alina-jacob 
I noticed that was already updating that in another PR.

**Pull request merged**